numpy/core/src/multiarray/mapping.c

```diff
@@ -1435,113 +1436,58 @@ _get_field_view(PyArrayObject *arr, PyObject *ind, PyArrayObject **view)
-            if (name == NULL) {
-                /* only happens for strange sequence objects */
+        /* check the items are strings */
+        for (npy_intp i = 0; i < seqlen; i++) {
+            PyObject *item = PySequence_GetItem(ind, i);
+            if(item == NULL) {
                 PyErr_Clear();
-                Py_DECREF(fields);
-                Py_DECREF(names);
                 return -1;
             }
-
-            if (!PyBaseString_Check(name)) {
-                Py_DECREF(name);
-                Py_DECREF(fields);
-                Py_DECREF(names);
+            npy_bool is_string = PyBaseString_Check(item);
+            Py_DECREF(item);
+            if(!is_string) {
                 return -1;
             }
+        }
 
-            tup = PyDict_GetItem(PyArray_DESCR(arr)->fields, name);
-            if (tup == NULL){
-                PyObject *errmsg = PyUString_FromString("no field of name ");
-                PyUString_ConcatAndDel(&errmsg, name);
-                PyErr_SetObject(PyExc_ValueError, errmsg);
-                Py_DECREF(errmsg);
-                Py_DECREF(fields);
-                Py_DECREF(names);
-                return 0;
-            }
-            // disallow use of titles as index
-            if (PyTuple_Size(tup) == 3) {
-                PyObject *title = PyTuple_GET_ITEM(tup, 2);
-                int titlecmp = PyObject_RichCompareBool(title, name, Py_EQ);
-                if (titlecmp == 1) {
-                    // if title == name, we were given a title, not a field name
-                    PyErr_SetString(PyExc_KeyError,
-                                "cannot use field titles in multi-field index");
-                }
-                if (titlecmp != 0 || PyDict_SetItem(fields, title, tup) < 0) {
-                    Py_DECREF(title);
-                    Py_DECREF(name);
-                    Py_DECREF(fields);
-                    Py_DECREF(names);
-                    return 0;
-                }
-                Py_DECREF(title);
-            }
-            // disallow duplicate field indices
-            if (PyDict_Contains(fields, name)) {
-                PyObject *errmsg = PyUString_FromString(
-                                       "duplicate field of name ");
-                PyUString_ConcatAndDel(&errmsg, name);
-                PyErr_SetObject(PyExc_ValueError, errmsg);
-                Py_DECREF(errmsg);
-                Py_DECREF(fields);
-                Py_DECREF(names);
-                return 0;
-            }
-            if (PyDict_SetItem(fields, name, tup) < 0) {
-                Py_DECREF(name);
-                Py_DECREF(fields);
-                Py_DECREF(names);
-                return 0;
-            }
-            if (PyTuple_SetItem(names, i, name) < 0) {
-                Py_DECREF(fields);
-                Py_DECREF(names);
-                return 0;
+        /* Convert sequences into a list, which is what dtype.__getitem__
+           expects. */
```

Comment:
Here I'm arbitrarily deciding that `descr.__getitem__` is going to be strict about the sequence type it gets passed, since it has no compatibility concerns. However, we still need to be liberal in what we accept here in `array.__getitem__`.